rdflib/plugins/sparql/parserutils.py

```diff
@@ -241,16 +228,14 @@ def postParse(self, instring, loc, tokenList):
                 res["service_string"] = service_string
 
         for t in tokenList:
-            if isinstance(t, ParamValue):
-                if t.isList:
-                    if t.name not in res:
-                        res[t.name] = plist()
-                    res[t.name].append(t.tokenList)
-                else:
-                    res[t.name] = t.tokenList
-                # res.append(t.tokenList)
-            # if isinstance(t,CompValue):
-            #    res.update(t)
+            if isinstance(t, ParseResults):
+                for i in t:
+                    if isinstance(i, ParamValue):
```

Comment:
Could it make sense to raise an exception if this condition does not match?

Reply:
Yeah I think that's a lot clearer, as otherwise it looks like we're filtering our for `ParamValue`s, but they should all be `ParamValue`s. Fixed in latest commit.

Reply:
Raising a TypeError here conflicts with this line in the docstring: "Any sub-tokens that are not Params will be ignored." So raising TypeError here may start breaking some existing code.

Reply:
I completely missed that! I'm not sure how true that line in the docstring is, but I think it's worth an investigation in a separate PR.